### PR TITLE
Migrate from vue cli to vite

### PR DESCRIPTION
**Pull request by zusorio**
This PR migrates the slmn.gg frontend from vue-cli to vite by using [this guide](https://vueschool.io/articles/vuejs-tutorials/how-to-migrate-from-vue-cli-to-vite/).

The main changes are:
- Environment variables now start with `VITE_` instead of `VUE_APP_`
- Environment variables are accessed from `import.meta.env` instead of `process.env`
- Babel was removed

### CODE REVIEW

**Comment by slmnio** [comment]
This looks good, I've had a play with it to try and break it and everything looks like it should be in working order. I might try to run it on the main server first, then we can merge it in completely.

File: website/package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "notyf": "^3.10.0",
     "obs-websocket-js": "^5.0.2",
     "opus-decoder": "^0.5.3",
+    "rollup-plugin-node-builtins": "^2.1.2",
```
Comment (slmnio) [inline website/package.json:21]: This is based on this issue I found when running it on the server: https://github.com/nuxt/vite/issues/160#issuecomment-983080874

**Comment by slmnio** [comment]
I used the build-stage action on the server, so [dev.slmn.gg](https://dev.slmn.gg) is live with it right now. It looks like Vue doesn't know that it's in developer mode, but the site does. I wonder if the stage building process can tell Vue to build in development mode so things are still accessible compared to the full prod build.
image

File: website/package.json (hunk 1 above)
Comment (slmnio) [reply website/package.json:21]: Note: not actually sure if this helped. I think it's something to do with the Jimp import here which tries to use the fs module on web. Having problems linked to this when trying to access anything with the RecoloredHero component
https://github.com/slmnio/slmngg-sfc/blob/5a97a86ecb99149a20f9db55eae0eac6abd94ddc/website/src/components/broadcast/RecoloredHero.vue#L18

**Comment by zusorio** [comment]
I've force-enabled vue devtools in staging in 80899b54234a6cbe4aea2110c13783b9f427d625, i can take a look at fs next

**Comment by slmnio** [comment]
Nice stuff so far

File: website/package.json (hunk 1 above)
Comment (zusorio) [reply website/package.json:21]: This has been fixed by replacing jimp with image-js

**Comment by zusorio** [comment]
Recoloring works again after replacing jimp with image-js

**Comment by slmnio** [comment]
Looks great

**Review by slmnio — APPROVED** [review]
Sending it

**Pull request merged**